ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt

```diff
@@ -65,15 +65,21 @@ internal class JsClientEngine(
     private fun createWebSocket(
         urlString_capturingHack: String,
         headers: Headers
-    ): WebSocket = when (PlatformUtils.platform) {
-        Platform.Browser -> js("new WebSocket(urlString_capturingHack)")
-        else -> {
-            val ws_capturingHack = js("eval('require')('ws')")
-            val headers_capturingHack: dynamic = object {}
-            headers.forEach { name, values ->
-                headers_capturingHack[name] = values.joinToString(",")
+    ): WebSocket {
+        val (protocolHeaderNames, otherHeaderNames) = headers.names().partition {
```

Comment:
why `partition` and not just `filter`?

Reply:
That may be an artifact of the apollographql specific work around (there is an instance where the other headers are examined in their code, I'll double check this here to see if it still makes sense, but filter would probably be good enough)